dftimewolf/lib/collectors/aws.py

```diff
@@ -0,0 +1,255 @@
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+
+
+class AWSCollector(module.BaseModule):
+  """Amazon Web Services (AWS) Collector.
+
+  Attributes:
+    remote_profile_name (str): The AWS account in which the volume(s) exist(s).
+        This is the profile name that is defined in your AWS credentials file.
+    remote_zone (str): The AWS zone in which the source volume(s) exist(s).
+    source_account (AWSAccount): The AWSAccount object that represents the
+        source account in which the volume(s) exist(s).
+    incident_id (str): Incident identifier used to name the analysis VM.
+    remote_instance_id (str): Instance ID that needs forensicating.
+    volume_ids (list[str]): List of volume ids to copy.
+    all_volumes (bool): True if all volumes attached to the source
+        instance should be copied.
+    analysis_profile_name (str): The AWS account in which to create the
+        analysis VM. This is the profile name that is defined in your AWS
+        credentials file.
+    analysis_zone (str): The AWS zone in which to create the VM.
+    analysis_vm (AWSInstance): Analysis VM to which the volume copy will be
+        attached.
+    device_names (list[str]): A list of available device names to be
+        used by AWS to attach volumes to the analysis VM.
+  """
+
+  _ANALYSIS_VM_CONTAINER_ATTRIBUTE_NAME = 'Analysis VM'
+  _ANALYSIS_VM_CONTAINER_ATTRIBUTE_TYPE = 'text'
+
+  def __init__(self, state, critical=False):
+    """Initializes an Amazon Web Services (AWS) collector.
+
+    Args:
+      state (DFTimewolfState): recipe state.
+      critical (Optional[bool]): True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super(AWSCollector, self).__init__(state, critical=critical)
+    self.remote_profile_name = None
+    self.remote_zone = None
+    self.source_account = None
+    self.incident_id = None
+    self.remote_instance_id = None
+    self.volume_ids = []
+    self.all_volumes = False
+    self.analysis_profile_name = None
+    self.analysis_zone = None
+    self.analysis_vm = None
+    self.device_suffixes = None
+
+  def Process(self):
+    """Copies a volume and attaches it to the analysis VM."""
+    for volume in self._FindVolumesToCopy():
+      print('Volume copy of {0:s} started...'.format(volume.volume_id))
+      new_volume = aws_forensics.CreateVolumeCopy(
+          self.remote_zone,
+          dst_zone=self.analysis_zone,
+          volume_id=volume.volume_id,
+          src_profile=self.remote_profile_name,
+          dst_profile=self.analysis_profile_name)
+      self.analysis_vm.AttachVolume(
+          new_volume, self._FindNextAvailableDeviceName())
+      print('Volume {0:s} successfully copied to {1:s}'.format(
+          volume.volume_id, new_volume.volume_id))
+      self.state.output.append((self.analysis_vm.name, new_volume))
```

Comment:
Since #279 we're not using input / output anymore. Please replace this with a a `ForensicsVM` attribute container like in [gcloud.py](https://github.com/log2timeline/dftimewolf/blob/master/dftimewolf/lib/collectors/gcloud.py#L70)